src/main/java/dev/jbang/cli/Export.java

```diff
@@ -502,3 +498,239 @@ private Path getJlinkOutputPath() {
+
+	abstract void renderBuildFile(BuildContext ctx, Path projectDir, String fullClassName) throws IOException;
+}
+
+@Command(name = "gradle", description = "Exports a Gradle project")
+class ExportGradleProject extends BaseExportProject {
+
+	@Override
+	String getType() {
+		return "gradle";
+	}
+
+	@Override
+	void renderBuildFile(BuildContext ctx, Path projectDir, String fullClassName) throws IOException {
+		Project prj = ctx.getProject();
+		ResourceRef templateRef = ResourceRef.forResource("classpath:/export-build.qute.gradle");
+		Path destination = projectDir.resolve("build.gradle");
+
+		List<MavenRepo> repositories = prj.getRepositories();
+		List<String> dependencies = prj.getMainSourceSet().getDependencies();
+		// Turn any URL dependencies into regular GAV coordinates
+		List<String> depIds = dependencies	.stream()
+											.map(JitPackUtil::ensureGAV)
+											.collect(Collectors.toList());
+		// And if we encountered URLs let's make sure the JitPack repo is available
+		if (!depIds.equals(dependencies)
+				&& repositories.stream().noneMatch(r -> DependencyUtil.REPO_JITPACK.equals(r.getUrl()))) {
+			prj.addRepository(DependencyUtil.toMavenRepo(DependencyUtil.ALIAS_JITPACK));
+		}
+
+		TemplateEngine engine = TemplateEngine.instance();
+		Template template = engine.getTemplate(templateRef);
+		if (template == null)
+			throw new ExitException(EXIT_INVALID_INPUT, "Could not locate template named: '" + templateRef + "'");
+		String result = template
+								.data("group", group)
+								.data("artifact", artifact)
+								.data("version", version)
+								.data("description", prj.getDescription().orElse(""))
+								.data("repositories", repositories	.stream()
+																	.map(MavenRepo::getUrl)
+																	.filter(s -> !"".equals(s))
+																	.collect(Collectors.toList()))
+								.data("javaVersion", prj.getJavaVersion())
+								.data("gradledependencies", gradleify(depIds))
+								.data("fullClassName", fullClassName)
+								.render();
+		Util.writeString(destination, result);
+	}
+
+	private List<String> gradleify(List<String> collectDependencies) {
+		return collectDependencies.stream().map(item -> {
+			if (item.endsWith("@pom")) {
+				return "implementation platform ('" + item.substring(0, item.lastIndexOf("@pom")) + "')";
+			} else {
+				return "implementation '" + item + "'";
+			}
+		}).collect(Collectors.toList());
+	}
+}
+
+@Command(name = "maven", description = "Exports a Maven project")
+class ExportMavenProject extends BaseExportProject {
+
+	@Override
+	String getType() {
+		return "maven";
+	}
+
+	@Override
+	void renderBuildFile(BuildContext ctx, Path projectDir, String fullClassName) throws IOException {
+		Project prj = ctx.getProject();
+		ResourceRef templateRef = ResourceRef.forResource("classpath:/export-pom.qute.xml");
+		Path destination = projectDir.resolve("pom.xml");
+
+		Map<String, String> properties = new HashMap<>();
+		if (prj.getJavaVersion() != null) {
```

Comment:
default jbang java is 8...should we explicilty set it?

Reply:
While it totally makes sense for building JBang itself, I'd guess Java 8 is a bit too restrictive for normal JBang users. I'd imagine most users just expect the default local Java version to be used unless they explicitly set `//JAVA`. I think this can be achieve just by using `JavaUtil.javaVersion`. Let me update the code accordingly and make it always set a Java version.

Reply:
Ah, thanks for your explanation. So, I agree that we should only set the version if the script explicitly set `//JAVA`. As Tako said, it seems to be really an edge case where the user fails with it.

I'll go ahead and change the PR again so that it strictly follows `//JAVA` and when absent it won't set the Java versions.

Reply:
i'm -0 on it but lets just see - we can change it later if problematic :)